src/filter/index.js

```diff
@@ -17,10 +17,79 @@ export default async function filter(content, selector, filterNames, filterFunct
     });
   }
 
-  const selectedContent = Array.from(webPageDOM.querySelectorAll(selector));
-  if (!selectedContent.length) {
-    console.warn(`The provided selector "${selector}" has no match in the web page.`);
+  let selectedContent = webPageDOM;
+
+  if (removeElements) {
+    let elementsToRemove = removeElements;
+
+    if (!Array.isArray(elementsToRemove)) {
+      elementsToRemove = [elementsToRemove];
+    }
+
+    elementsToRemove.forEach((elementSelector) => {
+      if (typeof elementSelector === 'object') {
+        const rangeSelection = getRangeSelection(selectedContent, elementSelector);
+        rangeSelection.deleteContents();
+      } else {
+        Array.from(webPageDOM.querySelectorAll(elementSelector)).forEach(node => node.remove());
+      }
+    });
+  }
+
+  const selectedContents = [];
+  if (selector) {
+    let elementsToSelect = selector;
+
+    if (!Array.isArray(elementsToSelect)) {
+      elementsToSelect = [elementsToSelect];
+    }
+
+    elementsToSelect.forEach((elementSelector) => {
```

Comment:
What about instead:
```suggestion
    selectedContents = [].concat(elementsToSelect).reduce((selection, selector)  => {
```
? 🙂